### PR TITLE
Fix a bug in `chalk` when `supports-color` gives level 2

### PR DESCRIPTION
**Pull request by partheseas**
The bug seems mainly apparent on Windows systems. (See https://github.com/chalk/chalk/issues/224)

### CODE REVIEW

**Comment by Qix-** [comment]
Any chance you could explain what exactly this fixes? I don't see how this would be causing a Windows-specific bug.

**Comment by partheseas** [comment]
I'm not sure if it's Windows specific, but Windows is the only operating system that I have access too and the only other people I could find with this issue have also been on Windows.

I did this bug fix in about an hour, and I probably dont understand the code base as well as you, but it seems to me that this is the issue:

When supports-color returns a level of 2 (ansi256) and you call `chalk.ansi256( int )( string )` it then internally calls `ansiStyles.color.ansi256.ansi256( int )`, which doesn't exist because why would you want to convert ansi256 to itself?

Setting --color=16m fixes this issue in a hacky-ish way because chalk will then call `ansiStyles.color.ansi16m.ansi256( int )` which exists, because you might want to reasonably convert between those two values.

There is an open issue on chalk that shows screenshots of what exactly the issue looks like when it happens. I provided a link in the description of the pull request.

**Comment by partheseas** [comment]
I just remembered about a Fedora virtual machine I had set up that I don't ever use. Tested it out and it seems that the issue is can be reproduced on other platforms where color-support returns with level 2 or by setting --color=256.
Fedora Screenshot

**Comment by partheseas** [comment]
@Qix- Is this a fix that can be implemented or do I need to look elsewhere for coloring my terminal output?

**Comment by Qix-** [comment]
Aha! I see what you mean now. Yes - can you add a failing test that reproduces this issue? :)

**Comment by partheseas** [comment]
I'm not exactly sure what the best way to test it would be since..
- We need to spoof color support
- The issue doesn't really surface in ansi-styles, it surfaces in Chalk

Any suggestions?

File: package.json [hunk 1]
```diff
@@ -41,7 +41,8 @@
 		"text"
 	],
 	"dependencies": {
-		"color-convert": "^1.9.0"
+		"color-convert": "^1.9.0",
+		"resolve-from": "^4.0.0"
```
Comment (sindresorhus) [inline package.json]: This should be in `devDependencies`.

**Comment by sindresorhus** [comment]
@partheseas Can you fix the merge conflict?

**Comment by partheseas** [comment]
@sindresorhus Yeah, I can fix the conflict. I'm still at a total loss about testing though. Should I leave the test failing? Remove the testing for it all together?

**Comment by sindresorhus** [comment]
Fixed by #46

**Pull request closed without merging**